### PR TITLE
Add displacement argument to the Lyapunov exponent computation

### PR DESCRIPTION
**Pull request by SebastianM-C**

### CODE REVIEW

File: src/lyapunovs.jl [hunk 1]
```diff
@@ -254,7 +254,7 @@ function lyapunov(ds::ContinuousDynamicalSystem, T = 10000.0; Ttr = 0.0,
   st1 = ds.state
   integ1 = ODEIntegrator(ds, T; diff_eq_kwargs=diff_eq_kwargs)
   integ1.opts.advance_to_tstop=true
-  ds.state = st1 .+ d0
+  ds.state = st1 .+ displacement(st1, d0)
```
Comment (Datseris) [inline src/lyapunovs.jl]: This is also incorrect.

File: src/lyapunovs.jl [hunk 2]
```diff
@@ -254,7 +254,7 @@ function lyapunov(ds::ContinuousDynamicalSystem, T = 10000.0; Ttr = 0.0,
   st1 = ds.state
   integ1 = ODEIntegrator(ds, T; diff_eq_kwargs=diff_eq_kwargs)
   integ1.opts.advance_to_tstop=true
-  ds.state = st1 .+ displacement(st1, d0)
+  ds.state = displacement.(st1, d0)
```
Comment (Datseris) [inline src/lyapunovs.jl]: I think the dot should be at the equals, i.e. `.= displacement`

Now, the function itself definitely doesn't need a dot. It doesn't operate element-wise on an array, but takes an array and produces a new one.

So `ds.state .= displacement(st1, d0)`

Comment (Datseris) [reply src/lyapunovs.jl]: I mean, in many cases the user will specify a function that would not make sense to be broadcasted. This sacrifices some speed but gives flexibility.

Plus this part of the code is not the bottleneck so speed wouldn't matter here.

File: src/lyapunovs.jl [hunk 3]
```diff
@@ -213,20 +214,20 @@ end
...
-  (state2, state1, d0) -> broadcast!(+, state2, state1, d0/sqrt(D))
+  sqD = sqrt(dimension(ds))
+  (state2, state1, d0) -> broadcast!(+, state2, state1, d0/sqD)
 end
 
 function lyapunov(ds::ContinuousDynamicalSystem,
-                  T::Real = 10000.0,
-                  return_convergence::Val{B} = Val{false};
+                  T::Real,
+                  return_convergence::Val{B} = Val{false}();
```
Comment (ChrisRackauckas) [inline src/lyapunovs.jl]: Don't need the `()` if you do `return_convergence::Type{Val{B}}` which is better accepted.

**Comment by Datseris** [comment]
You can know actually use `lyapunov` finally.

However, now it gives wrong results:
```
ds = Systems.lorenz()
lyapunov(ds, 10000.0)
```
gives 1.0 ... instead of 0.9 ...

I dont know whats wrong but it has to be fixed before merging..

**Comment by Datseris** [comment]
BOOOM Test pass this NIGHTMARE can finally be merged after draining my almost empty soul.

@SebastianM-C  I have changed completely the interface so you might want to study the source code before you use it.

**Pull request merged**